### PR TITLE
Ruby on Rails capstone project - Budget app

### PR DESCRIPTION
**Pull request by abdullah-asghar-khan**
https://www.loom.com/share/56dde45beb7f498c81178bb9597a9f2e

**Interactions**

**Splash screen**

- A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.

**Sign up and log in pages**

- The user should be able to register in the app with full name, email and password (all mandatory).

- The user can log into the app using email and password.

- If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

**Homepage (catagories page)**

- When the user logs in, they are presented with the categories page.

- For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.

- When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.

- There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

**Transactions page**

- For a given category, the list of transactions is presented, ordered by the most recent.

- At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).

- There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.

- When the user clicks on the "Back" button (<), the user navigates to the home page.

**"Add a new category" page**

- The user fills out a form to create a new category, indicating their name and icon (both mandatory).

- The user click (or taps) the "Save" button to create the new category, and is taken to the home page on success.

- When the user clicks on the "Back" button (<), the user navigates to the home page.

**"Add a new transaction" page**

- The user fills out a form to create a new transaction with:

- name (mandatory)

- amount (mandatory)

- categories (mandatory at least one)

- The user click (or taps) the "Save" button to create the new transation, and is taken to the transactions page for that category.

- When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

**Testing requirements**

- Implemented unit and integration tests for all the most important components of your RoR application.

**Technical requirements**

- I used the following as required:

- Postgres for database.

- devise for authentication.

- validate all user input to make sure that anyone with bad intentions cannot compromise your app.

- .erb view template engine.

- Deployed the project on Heroku and it is accessible online.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,10 +11,10 @@
 - Bootstap
 
 ## Video Describing the project features
```
Comment (SajjadAhmad14) [inline README.md:13]: Thank you for your presentation. Kindly record a new video which only highlights a piece of code which you are proud of and find interesting.

Please add integretion specs to test if all the parts of the app work together without any issues.

Please use the fonts and colors already defined in the design page for the whole project. The `Add Category` button should have `green` color and it should be fixed on the bottom of the page.

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Abdullah, 
 ### You have been doing a tremendous job so far :ok_hand:

## To Highlight 🎉
Git fow is followed. :heavy_check_mark:
README is professional. :heavy_check_mark:
## To Improve 🙏

- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
**_Note: Please do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: README.md (hunk 1 above)
Comment (abdullah-asghar-khan) [reply README.md:13]: Hi @SajjadAhmad14, the add category has already green color and it´s on the bottom of the page :)

**Review by mukhammadrizooff — APPROVED** [review]
Hi @Abdullah2213565, 

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 



## Highlights 
- Clear work ✔️
- No linter errors ✔️  
- Good implementations ✔️ 
- Descriptive  commit history ✔️
- Descriptive `PR` description ✔️
 
Good luck! 🎉 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**